# Practica Grupal 1

## Packages

In [1]:
library("dplyr")
library(data.table)


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union


Warning message:
“package ‘data.table’ was built under R version 3.6.3”

Attaching package: ‘data.table’


The following objects are masked from ‘package:dplyr’:

    between, first, last




## Load Data

In [2]:
df = read.csv2("datasets/SABERTYT20162.csv", sep = ";", dec = ",", encoding ="latin1", na.strings = c("NA",""))

In [3]:
df

ESTU_TIPODOCUMENTO ESTU_NACIONALIDAD ESTU_GENERO ESTU_FECHANACIMIENTO
1     CC                 1 COLOMBIA        M           15/08/1990          
2     CC                 1 COLOMBIA        M           26/09/1990          
3     CC                 1 COLOMBIA        F           28/10/1994          
4     CC                 1 COLOMBIA        F           24/09/1979          
5     CC                 1 COLOMBIA        M           6/07/1988           
6     CC                 1 COLOMBIA        M           31/12/1992          
7     CC                 1 COLOMBIA        F           19/04/1975          
8     CC                 1 COLOMBIA        M           1/04/1995           
9     CC                 1 COLOMBIA        M           20/12/1988          
10    CC                 1 COLOMBIA        F           30/12/1997          
11    CC                 1 COLOMBIA        M           5/03/1977           
12    CC                 1 COLOMBIA        F           21/01/1986          
13    CC                 1 COLOMBIA        F           11/11/1989          
14    CC                 1 COLOMBIA        F           6/06/1995           
15    CC                 1 COLOMBIA        M           9/11/1985           
16    CC                 1 COLOMBIA        F           20/04/1992          
17    CC                 1 COLOMBIA        F           5/07/1995           
18    CC                 1 COLOMBIA        M           8/06/1973           
19    CC                 1 COLOMBIA        M           29/01/1996          
20    CC                 1 COLOMBIA        M           3/01/1991           
21    CC                 1 COLOMBIA        F           15/05/1990          
22    CC                 1 COLOMBIA        M           12/08/1990          
23    CC                 1 COLOMBIA        M           29/07/1996          
24    TI                 1 COLOMBIA        F           29/04/1999          
25    CC                 1 COLOMBIA        M           14/10/1994          
26    CC                 1 COLOMBIA        F           24/01/1993          
27    CC                 1 COLOMBIA        M           23/08/1994          
28    CC                 1 COLOMBIA        F           15/02/1994          
29    CC                 1 COLOMBIA        M           12/05/1995          
30    CC                 1 COLOMBIA        M           13/07/1997          
⋮     ⋮                  ⋮                 ⋮           ⋮                   
53011 CC                 1 COLOMBIA        F           10/12/1993          
53012 CC                 1 COLOMBIA        M           30/10/1994          
53013 CC                 1 COLOMBIA        M           4/05/1996           
53014 TI                 1 COLOMBIA        M           6/12/1997           
53015 CC                 1 COLOMBIA        M           18/10/1984          
53016 CC                 1 COLOMBIA        M           27/10/1997          
53017 CC                 1 COLOMBIA        M           17/07/1995          
53018 CC                 1 COLOMBIA        M           11/08/1965          
53019 CC                 1 COLOMBIA        F           23/01/1993          
53020 CC                 1 COLOMBIA        F           16/03/1996          
53021 CC                 1 COLOMBIA        M           11/10/1983          
53022 CC                 1 COLOMBIA        F           19/10/1988          
53023 CC                 1 COLOMBIA        M           13/11/1993          
53024 CC                 1 COLOMBIA        M           10/09/1993          
53025 CC                 1 COLOMBIA        M           12/05/1983          
53026 CC                 1 COLOMBIA        M           8/01/1994           
53027 CC                 1 COLOMBIA        M           12/02/1994          
53028 CC                 1 COLOMBIA        F           1/09/1980           
53029 TI                 1 COLOMBIA        F           26/01/1999          
53030 CC                 1 COLOMBIA        F           3/04/1995           
53031 CC                 1 COLOMBIA        F          

## Filtrado de Datos

Nos quedaremos únicamente con los datos necesarios para los analisis.

### Resultados listos para publicar

Nos interesa realizar el estudio de los resultados listos para publicarse.

Revisamos los niveles de los datos de la variable ESTUD_ESTADO que nos indica el estado del resultado.

In [4]:
unique(df$ESTU_ESTADO)

[1] PUBLICAR                 VALIDEZ OFICINA JURÍDICA
Levels: PUBLICAR VALIDEZ OFICINA JURÍDICA

Nos damos cuenta que solo tenemos los 2 valores esperado por lo que procedemos a filtrar los publicados.

In [5]:
df_published = df[df$ESTU_ESTADO == "PUBLICAR",]

In [6]:
unique(df_published$ESTU_ESTADO)

[1] PUBLICAR
Levels: PUBLICAR VALIDEZ OFICINA JURÍDICA

Ahora solo tenemos los resultados listos para publicarse.

### Filtrado de Columnas

Solo necesitamos los datos de la columna 1 a la 15 y de la 85 a la 104 para el analisis, por lo que procedemos a retirar el resto de variables.

In [7]:
cols = c(1:15,85:104)
df_published = df_published[,cols]

In [8]:
df_published[c(1:5),]

ESTU_TIPODOCUMENTO ESTU_NACIONALIDAD ESTU_GENERO ESTU_FECHANACIMIENTO PERIODO
1 CC                 1 COLOMBIA        M           15/08/1990           20164  
2 CC                 1 COLOMBIA        M           26/09/1990           20164  
3 CC                 1 COLOMBIA        F           28/10/1994           20164  
4 CC                 1 COLOMBIA        F           24/09/1979           20164  
5 CC                 1 COLOMBIA        M           6/07/1988            20164  
  ESTU_CONSECUTIVO ESTU_ESTUDIANTE ESTU_ESTADOCIVIL ESTU_TIENE_ETNIA ESTU_ETNIA
1 EK201640024941   ESTUDIANTE      UNIÓN LIBRE      NO               NA        
2 EK201640012159   ESTUDIANTE      SOLTERO          NA               NA        
3 EK201640041435   ESTUDIANTE      SOLTERO          NO               NA        
4 EK201640003654   ESTUDIANTE      UNIÓN LIBRE      NO               NA        
5 EK201640018456   ESTUDIANTE      SOLTERO          NO               NA        
  ⋯ MOD_INGLES_DESEM MOD_INGLES_PNAL MOD_INGLES_PGREF MOD_COMUNI_ESCRITA_PUNT
1 ⋯ A2               79              59               121                    
2 ⋯ A2               75              74               123                    
3 ⋯ B1               97              98                99                    
4 ⋯ A1               32              30                75                    
5 ⋯ A1-              15              19                86                    
  MOD_COMUNI_ESCRITA_DESEM MOD_COMUNI_ESCRITA_PNAL MOD_COMUNI_ESCRITA_PGREF
1 4                        87                      85                      
2 4                        89                      87                      
3 2                        43                      44                      
4 1                        13                      11                      
5 2                        29                      34                      
  ESTU_ESTADO INSE     NSE 
1 PUBLICAR    57.14241 NSE3
2 PUBLICAR    65.76637 NSE4
3 PUBLICAR    47.68200 NSE2
4 PUBLICAR    55.56398 NSE3
5 PUBLICAR    27.01330 NSE1

### Nacionalidad

Solo necesitamos a los estudiantes de nacionalidad Colombiana por lo que descartamos a los que no tengan dicha nacionalidad.

Revisamos primero que no existan problemas en la forma que están escritas las nacionalidades.

In [9]:
unique(df_published$ESTU_NACIONALIDAD)

[1] 1 COLOMBIA              236 ESTADOS UNIDOS      241 VENEZUELA          
 [4] 50 COMORAS              46 CHILE                209 ESPAÑA             
 [7] 64 REPÚBLICA DOMINICANA 174 PERÚ                83 ALEMANIA            
[10] 167 OMÁN                119 COREA DEL SUR       67 EL SALVADOR         
[13] 55 COSTA DE MARFIL      65 ECUADOR              171 PANAMÁ             
[16] 3 ALAND                 118 COREA DEL NORTE     144 MÉXICO             
[19] 112 JAPÓN               96 HAITI               
21 Levels: 1 COLOMBIA 112 JAPÓN 118 COREA DEL NORTE ... 96 HAITI

Podemos ver que no hay nacionalidades mal escritas o que se encuentren repetidas pero escritas de diferentes formas.

Suponemos que el número al inicio de la nacionalidad es un identificador del país y no genera problemas para el estudio, sin embargo no estaría de más preguntar el porque este dato se encuentra de esta forma para estar seguros de que no se trata de un problema.

In [10]:
df_published = df_published[df_published$ESTU_NACIONALIDAD == "1 COLOMBIA",]

In [11]:
df_published[c(1:5),]

ESTU_TIPODOCUMENTO ESTU_NACIONALIDAD ESTU_GENERO ESTU_FECHANACIMIENTO PERIODO
1 CC                 1 COLOMBIA        M           15/08/1990           20164  
2 CC                 1 COLOMBIA        M           26/09/1990           20164  
3 CC                 1 COLOMBIA        F           28/10/1994           20164  
4 CC                 1 COLOMBIA        F           24/09/1979           20164  
5 CC                 1 COLOMBIA        M           6/07/1988            20164  
  ESTU_CONSECUTIVO ESTU_ESTUDIANTE ESTU_ESTADOCIVIL ESTU_TIENE_ETNIA ESTU_ETNIA
1 EK201640024941   ESTUDIANTE      UNIÓN LIBRE      NO               NA        
2 EK201640012159   ESTUDIANTE      SOLTERO          NA               NA        
3 EK201640041435   ESTUDIANTE      SOLTERO          NO               NA        
4 EK201640003654   ESTUDIANTE      UNIÓN LIBRE      NO               NA        
5 EK201640018456   ESTUDIANTE      SOLTERO          NO               NA        
  ⋯ MOD_INGLES_DESEM MOD_INGLES_PNAL MOD_INGLES_PGREF MOD_COMUNI_ESCRITA_PUNT
1 ⋯ A2               79              59               121                    
2 ⋯ A2               75              74               123                    
3 ⋯ B1               97              98                99                    
4 ⋯ A1               32              30                75                    
5 ⋯ A1-              15              19                86                    
  MOD_COMUNI_ESCRITA_DESEM MOD_COMUNI_ESCRITA_PNAL MOD_COMUNI_ESCRITA_PGREF
1 4                        87                      85                      
2 4                        89                      87                      
3 2                        43                      44                      
4 1                        13                      11                      
5 2                        29                      34                      
  ESTU_ESTADO INSE     NSE 
1 PUBLICAR    57.14241 NSE3
2 PUBLICAR    65.76637 NSE4
3 PUBLICAR    47.68200 NSE2
4 PUBLICAR    55.56398 NSE3
5 PUBLICAR    27.01330 NSE1

In [12]:
dim(df_published)

[1] 52665    35

Al final nos quedamos con un conjunto de datos de 52665 y de 35 columnas.

## Formateo de los Datos

Revisamos que los datos se encuentren en el formato correcto.

In [13]:
class(df_published)

[1] "data.frame"

Revisemos la clase de las de las variables

In [14]:
lapply(df_published, class)

$ESTU_TIPODOCUMENTO
[1] "factor"

$ESTU_NACIONALIDAD
[1] "factor"

$ESTU_GENERO
[1] "factor"

$ESTU_FECHANACIMIENTO
[1] "factor"

$PERIODO
[1] "integer"

$ESTU_CONSECUTIVO
[1] "factor"

$ESTU_ESTUDIANTE
[1] "factor"

$ESTU_ESTADOCIVIL
[1] "factor"

$ESTU_TIENE_ETNIA
[1] "factor"

$ESTU_ETNIA
[1] "factor"

$ESTU_LIMITA_MOTRIZ
[1] "factor"

$ESTU_LIMITA_INVIDENTE
[1] "factor"

$ESTU_LIMITA_CONDICIONESPECIAL
[1] "factor"

$ESTU_LIMITA_SORDO
[1] "factor"

$ESTU_LIMITA_AUTISMO
[1] "factor"

$MOD_RAZONA_CUANTITAT_PUNT
[1] "integer"

$MOD_RAZONA_CUANTITATIVO_PNAL
[1] "integer"

$MOD_RAZONA_CUANTITATIVO_PGREF
[1] "integer"

$MOD_LECTURA_CRITICA_PUNT
[1] "integer"

$MOD_LECTURA_CRITICA_PNAL
[1] "integer"

$MOD_LECTURA_CRITICA_PGREF
[1] "integer"

$MOD_COMPETEN_CIUDADA_PUNT
[1] "integer"

$MOD_COMPETEN_CIUDADA_PNAL
[1] "integer"

$MOD_COMPETEN_CIUDADA_PGREF
[1] "integer"

$MOD_INGLES_PUNT
[1] "integer"

$MOD_INGLES_DESEM
[1] "factor"

$MOD_INGLES_PNAL
[1] "integer"

$MOD_INGLES_PGREF
[1] "integer"

$MOD_COMUNI_ESCRITA_PUNT
[1] "integer"

$MOD_COMUNI_ESCRITA_DESEM
[1] "integer"

$MOD_COMUNI_ESCRITA_PNAL
[1] "integer"

$MOD_COMUNI_ESCRITA_PGREF
[1] "integer"

$ESTU_ESTADO
[1] "factor"

$INSE
[1] "numeric"

$NSE
[1] "factor"

Todas las variables menos la fecha de nacimiento, ESTU_TIENE_ETNIA, las limitaciones, MOD_COMUNI_ESCRITA_DESEM y el periodo se encuentran en un formato incorrecto ya que la primera se trata de una fecha y los demás de datos booleanos que se encuentran representados por un SI o NO o una X. Estos datos booleanos se recomienda realizar una estandarización como el uso del 0 y 1 ya que representan datos booleanos.

### Periodo

In [15]:
class(df_published$PERIODO)

[1] "integer"

In [16]:
df_published$PERIODO = as.factor(df_published$PERIODO)

### MOD_COMUNI_ESCRITA_DESEM

In [17]:
class(df_published$MOD_COMUNI_ESCRITA_DESEM)

[1] "integer"

In [18]:
df_published$MOD_COMUNI_ESCRITA_DESEM = as.factor(df_published$MOD_COMUNI_ESCRITA_DESEM)

In [19]:
unique(df_published$MOD_COMUNI_ESCRITA_DESEM)

[1] 4    2    1    3    <NA>
Levels: 1 2 3 4

### Fecha

In [20]:
class(df_published$ESTU_FECHANACIMIENTO)

[1] "factor"

Volveremos la fecha al formato correcto.

In [21]:
df_published$ESTU_FECHANACIMIENTO[c(1:5)]

[1] 15/08/1990 26/09/1990 28/10/1994 24/09/1979 6/07/1988 
10104 Levels: 1/01/1956 1/01/1972 1/01/1974 1/01/1975 1/01/1976 ... 9/12/2005

In [22]:
df_published$ESTU_FECHANACIMIENTO = as.character(df_published$ESTU_FECHANACIMIENTO)

In [23]:
df_published$ESTU_FECHANACIMIENTO = as.Date(df_published$ESTU_FECHANACIMIENTO, format = "%d/%m/%Y")

In [24]:
df_published$ESTU_FECHANACIMIENTO[c(1:5)]

[1] "1990-08-15" "1990-09-26" "1994-10-28" "1979-09-24" "1988-07-06"

In [25]:
class(df_published$ESTU_FECHANACIMIENTO)

[1] "Date"

Ya tenemos la fecha en el formato correcto

### Etnia

Estandaricemos la respuesta de si una persona pertenece a algún grupo etnico o no.

In [26]:
unique(df_published$ESTU_TIENE_ETNIA)

[1] NO   <NA> SI  
Levels: NO SI

Podemos ver que solo cuenta con SI No o valores nulos. Entonces volveremos los si a TRUE y los no a FALSE.

In [27]:
df_published$ESTU_TIENE_ETNIA = as.character(df_published$ESTU_TIENE_ETNIA)

In [28]:
df_published = df_published %>% mutate(ESTU_TIENE_ETNIA = replace(ESTU_TIENE_ETNIA, ESTU_TIENE_ETNIA == "NO", FALSE))

In [29]:
df_published = df_published %>% mutate(ESTU_TIENE_ETNIA = replace(ESTU_TIENE_ETNIA, ESTU_TIENE_ETNIA == "SI", TRUE))

In [30]:
class(df_published$ESTU_TIENE_ETNIA)

[1] "character"

In [31]:
unique(df_published$ESTU_TIENE_ETNIA)

[1] "FALSE" NA      "TRUE"

In [32]:
df_published$ESTU_TIENE_ETNIA = as.logical(df_published$ESTU_TIENE_ETNIA)

In [33]:
class(df_published$ESTU_TIENE_ETNIA)

[1] "logical"

### Limitaciones

Ahora revisemos las limitaciones que en su mayoría se encuentran con un formato poco entendible que es una x

In [34]:
colnames(df_published)[c(11:15)]

[1] "ESTU_LIMITA_MOTRIZ"            "ESTU_LIMITA_INVIDENTE"        
[3] "ESTU_LIMITA_CONDICIONESPECIAL" "ESTU_LIMITA_SORDO"            
[5] "ESTU_LIMITA_AUTISMO"

In [35]:
unique(df_published$ESTU_LIMITA_MOTRIZ)

[1] <NA> x   
Levels: x

In [36]:
lapply(df_published[c(11:15)], unique)

$ESTU_LIMITA_MOTRIZ
[1] <NA> x   
Levels: x

$ESTU_LIMITA_INVIDENTE
[1] <NA> x   
Levels: x

$ESTU_LIMITA_CONDICIONESPECIAL
[1] <NA> x   
Levels: x

$ESTU_LIMITA_SORDO
[1] <NA> x   
Levels: x

$ESTU_LIMITA_AUTISMO
[1] <NA> x   
Levels: x

Normalmente cuando se pregunta si una persona tiene discapacidad de le pide que llene marque una X, en este caso no sabemos si la X se refiere a una respuesta afirmativa o negativa

 Sin embargo podemos revisar el número de x marcadas y de campos dejados en blanco.

In [37]:
df_published %>% group_by(ESTU_LIMITA_MOTRIZ) %>% summarize(count=n())

ESTU_LIMITA_MOTRIZ count
1 x                     46
2 NA                 52619

In [38]:
df_published %>% group_by(ESTU_LIMITA_INVIDENTE) %>% summarize(count=n())

ESTU_LIMITA_INVIDENTE count
1 x                         4
2 NA                    52661

In [39]:
df_published %>% group_by(ESTU_LIMITA_CONDICIONESPECIAL) %>% summarize(count=n())

ESTU_LIMITA_CONDICIONESPECIAL count
1 x                                 7
2 NA                            52658

In [40]:
df_published %>% group_by(ESTU_LIMITA_SORDO) %>% summarize(count=n())

ESTU_LIMITA_SORDO count
1 x                    29
2 NA                52636

In [41]:
df_published %>% group_by(ESTU_LIMITA_AUTISMO) %>% summarize(count=n())

ESTU_LIMITA_AUTISMO count
1 x                       3
2 NA                  52662

In [42]:
unique(df_published$ESTU_LIMITA_AUTISMO)

[1] <NA> x   
Levels: x

Con esto podemos ver que las x son realmente personas con discapacidad puesto que representan un porcentaje minoritario.

Transformaremos los NA a FALSE y las X a TRUE

#### ESTU_LIMITA_MOTRIZ

In [43]:
df_published$ESTU_LIMITA_MOTRIZ = as.character(df_published$ESTU_LIMITA_MOTRIZ)

In [44]:
df_published$ESTU_LIMITA_MOTRIZ[is.na(df_published$ESTU_LIMITA_MOTRIZ)] = FALSE

In [45]:
df_published$ESTU_LIMITA_MOTRIZ[df_published$ESTU_LIMITA_MOTRIZ == 'x'] = TRUE

In [46]:
df_published$ESTU_LIMITA_MOTRIZ = as.logical(df_published$ESTU_LIMITA_MOTRIZ)

In [47]:
class(df_published$ESTU_LIMITA_MOTRIZ)

[1] "logical"

In [48]:
unique(df_published$ESTU_LIMITA_MOTRIZ)

[1] FALSE  TRUE

#### ESTU_LIMITA_INVIDENTE

In [49]:
df_published$ESTU_LIMITA_INVIDENTE = as.character(df_published$ESTU_LIMITA_INVIDENTE)

In [50]:
df_published$ESTU_LIMITA_INVIDENTE[is.na(df_published$ESTU_LIMITA_INVIDENTE)] = FALSE

In [51]:
df_published$ESTU_LIMITA_INVIDENTE[df_published$ESTU_LIMITA_INVIDENTE == 'x'] = TRUE

In [52]:
df_published$ESTU_LIMITA_INVIDENTE = as.logical(df_published$ESTU_LIMITA_INVIDENTE)

In [53]:
class(df_published$ESTU_LIMITA_INVIDENTE)

[1] "logical"

In [54]:
unique(df_published$ESTU_LIMITA_INVIDENTE)

[1] FALSE  TRUE

#### ESTU_LIMITA_CONDICIONESPECIAL

In [55]:
df_published$ESTU_LIMITA_CONDICIONESPECIAL = as.character(df_published$ESTU_LIMITA_CONDICIONESPECIAL)
df_published$ESTU_LIMITA_CONDICIONESPECIAL[is.na(df_published$ESTU_LIMITA_CONDICIONESPECIAL)] = FALSE
df_published$ESTU_LIMITA_CONDICIONESPECIAL[df_published$ESTU_LIMITA_CONDICIONESPECIAL == 'x'] = TRUE
df_published$ESTU_LIMITA_CONDICIONESPECIAL = as.logical(df_published$ESTU_LIMITA_CONDICIONESPECIAL)

In [56]:
class(df_published$ESTU_LIMITA_INVIDENTE)

[1] "logical"

In [57]:
unique(df_published$ESTU_LIMITA_INVIDENTE)

[1] FALSE  TRUE

#### ESTU_LIMITA_SORDO

In [58]:
df_published$ESTU_LIMITA_SORDO = as.character(df_published$ESTU_LIMITA_SORDO)
df_published$ESTU_LIMITA_SORDO[is.na(df_published$ESTU_LIMITA_SORDO)] = FALSE
df_published$ESTU_LIMITA_SORDO[df_published$ESTU_LIMITA_SORDO == 'x'] = TRUE
df_published$ESTU_LIMITA_SORDO = as.logical(df_published$ESTU_LIMITA_SORDO)

In [59]:
class(df_published$ESTU_LIMITA_SORDO)

[1] "logical"

In [60]:
unique(df_published$ESTU_LIMITA_SORDO)

[1] FALSE  TRUE

#### ESTU_LIMITA_AUTISMO

In [61]:
df_published$ESTU_LIMITA_AUTISMO = as.character(df_published$ESTU_LIMITA_AUTISMO)
df_published$ESTU_LIMITA_AUTISMO[is.na(df_published$ESTU_LIMITA_AUTISMO)] = FALSE
df_published$ESTU_LIMITA_AUTISMO[df_published$ESTU_LIMITA_AUTISMO == 'x'] = TRUE
df_published$ESTU_LIMITA_AUTISMO = as.logical(df_published$ESTU_LIMITA_AUTISMO)

In [62]:
class(df_published$ESTU_LIMITA_AUTISMO)

[1] "logical"

In [63]:
unique(df_published$ESTU_LIMITA_AUTISMO)

[1] FALSE  TRUE

In [64]:
lapply(df_published[c(11:15)], unique)

$ESTU_LIMITA_MOTRIZ
[1] FALSE  TRUE

$ESTU_LIMITA_INVIDENTE
[1] FALSE  TRUE

$ESTU_LIMITA_CONDICIONESPECIAL
[1] FALSE  TRUE

$ESTU_LIMITA_SORDO
[1] FALSE  TRUE

$ESTU_LIMITA_AUTISMO
[1] FALSE  TRUE

Ya tenemos los campos de limitaciones estandarizados

In [65]:
lapply(df_published, class)

$ESTU_TIPODOCUMENTO
[1] "factor"

$ESTU_NACIONALIDAD
[1] "factor"

$ESTU_GENERO
[1] "factor"

$ESTU_FECHANACIMIENTO
[1] "Date"

$PERIODO
[1] "factor"

$ESTU_CONSECUTIVO
[1] "factor"

$ESTU_ESTUDIANTE
[1] "factor"

$ESTU_ESTADOCIVIL
[1] "factor"

$ESTU_TIENE_ETNIA
[1] "logical"

$ESTU_ETNIA
[1] "factor"

$ESTU_LIMITA_MOTRIZ
[1] "logical"

$ESTU_LIMITA_INVIDENTE
[1] "logical"

$ESTU_LIMITA_CONDICIONESPECIAL
[1] "logical"

$ESTU_LIMITA_SORDO
[1] "logical"

$ESTU_LIMITA_AUTISMO
[1] "logical"

$MOD_RAZONA_CUANTITAT_PUNT
[1] "integer"

$MOD_RAZONA_CUANTITATIVO_PNAL
[1] "integer"

$MOD_RAZONA_CUANTITATIVO_PGREF
[1] "integer"

$MOD_LECTURA_CRITICA_PUNT
[1] "integer"

$MOD_LECTURA_CRITICA_PNAL
[1] "integer"

$MOD_LECTURA_CRITICA_PGREF
[1] "integer"

$MOD_COMPETEN_CIUDADA_PUNT
[1] "integer"

$MOD_COMPETEN_CIUDADA_PNAL
[1] "integer"

$MOD_COMPETEN_CIUDADA_PGREF
[1] "integer"

$MOD_INGLES_PUNT
[1] "integer"

$MOD_INGLES_DESEM
[1] "factor"

$MOD_INGLES_PNAL
[1] "integer"

$MOD_INGLES_PGREF
[1] "integer"

$MOD_COMUNI_ESCRITA_PUNT
[1] "integer"

$MOD_COMUNI_ESCRITA_DESEM
[1] "factor"

$MOD_COMUNI_ESCRITA_PNAL
[1] "integer"

$MOD_COMUNI_ESCRITA_PGREF
[1] "integer"

$ESTU_ESTADO
[1] "factor"

$INSE
[1] "numeric"

$NSE
[1] "factor"

Ya tenemos los valores en sus formatos correctos por lo que procederemos a ver la calidad de los datos.

## Calidad de los Datos

### Fecha

Todos los datos corresponden al 2016

In [66]:
unique(df_published$PERIODO)

[1] 20164
Levels: 20164

Podemos notar que hay fechas que han sido mal tomadas puesto que tienen fecha del 2024 cuando actualmente estamos en el 2021, es decir que hay personas que no han nacido en los datos. Este dato debería obtenerse desde una fuente de confianza como la registraduría y evitar que sean las personas quienes llenen estos datos. 

In [67]:
min(df_published$ESTU_FECHANACIMIENTO, na.rm = TRUE)
max(df_published$ESTU_FECHANACIMIENTO, na.rm = TRUE)

[1] "1944-12-13"

[1] "2024-01-16"

Lo recomendable para esta situación es solicitar los datos a la registraduría o a una fuente que nos permita obtener la fecha de nacimiento con total certeza y sin errores.

In [68]:
df_published[df_published$ESTU_FECHANACIMIENTO>as.Date("2000-12-31"),]

ESTU_TIPODOCUMENTO ESTU_NACIONALIDAD ESTU_GENERO ESTU_FECHANACIMIENTO
387   TI                 1 COLOMBIA        F           2001-04-16          
406   TI                 1 COLOMBIA        F           2009-08-23          
696   CC                 1 COLOMBIA        F           2016-08-15          
831   CC                 1 COLOMBIA        M           2015-10-29          
1584  CC                 1 COLOMBIA        M           2016-05-14          
2192  CC                 1 COLOMBIA        F           2004-10-16          
5020  TI                 1 COLOMBIA        F           2001-01-16          
5063  CC                 1 COLOMBIA        F           2016-07-08          
5065  CC                 1 COLOMBIA        F           2016-08-09          
5153  CC                 1 COLOMBIA        F           2015-03-30          
5641  CC                 1 COLOMBIA        F           2016-01-26          
5667  TI                 1 COLOMBIA        F           2001-10-22          
6035  CC                 1 COLOMBIA        M           2016-03-03          
6355  CC                 1 COLOMBIA        M           2016-04-04          
6383  CC                 1 COLOMBIA        M           2002-04-22          
9221  CC                 1 COLOMBIA        M           2016-01-31          
11479 CC                 1 COLOMBIA        M           2016-05-18          
11801 CC                 1 COLOMBIA        F           2016-12-12          
12901 CC                 1 COLOMBIA        M           2015-10-18          
13024 TI                 1 COLOMBIA        F           2014-05-02          
14260 TI                 1 COLOMBIA        F           2001-05-06          
15084 CC                 1 COLOMBIA        M           2016-06-16          
15104 CC                 1 COLOMBIA        M           2002-12-03          
15559 CC                 1 COLOMBIA        F           2016-04-12          
16096 CC                 1 COLOMBIA        F           2004-08-30          
16268 CC                 1 COLOMBIA        F           2016-08-27          
16401 CC                 1 COLOMBIA        M           2016-07-25          
16839 CC                 1 COLOMBIA        M           2001-12-07          
17166 CC                 1 COLOMBIA        F           2016-03-06          
17435 TI                 1 COLOMBIA        F           2001-02-11          
⋮     ⋮                  ⋮                 ⋮           ⋮                   
37522 CC                 1 COLOMBIA        F           2016-08-17          
38122 TI                 1 COLOMBIA        M           2001-02-11          
38198 CC                 1 COLOMBIA        F           2016-01-16          
38495 CC                 1 COLOMBIA        M           2024-01-16          
39221 CC                 1 COLOMBIA        M           2016-07-20          
39377 CC                 1 COLOMBIA        F           2016-01-06          
41545 CC                 1 COLOMBIA        F           2016-07-18          
41646 TI                 1 COLOMBIA        M           2001-03-09          
41720 CC                 1 COLOMBIA        M           2004-12-04          
41854 CC                 1 COLOMBIA        F           2016-02-10          
43011 CC                 1 COLOMBIA        M           2016-07-11          
43731 CC                 1 COLOMBIA        F           2004-12-31          
43746 CC                 1 COLOMBIA        F           2016-08-02          
44528 TI                 1 COLOMBIA        F           2001-01-03          
45314 CC                 1 COLOMBIA        M           2016-02-21          
46059 CC                 1 COLOMBIA        F           2001-11-17          
47063 CC                 1 COLOMBIA        M           2015-10-25          
47074 TI                 1 COLOMBIA        F           2001-01-02          
47647 CC                 1 COLOMBIA        M           2016-01-23          
47952 CC                 1 COLOMBIA        M           2016-09-21          
49170 CC                 1 COLOMBIA        M          

Hay alrededor de 92 personas que tenían menos de 16 años antes de presentar la prueba y algunos que incluso no habían nacido.

También hay personas con más de 70 años que presentarón la prueba en el 2016 esto sería un dato interesante pero que no podemos confiar debido a la forma en la que se recolectó este campo.

In [69]:
df_published[df_published$ESTU_FECHANACIMIENTO < as.Date("1946-12-31"),]

ESTU_TIPODOCUMENTO ESTU_NACIONALIDAD ESTU_GENERO ESTU_FECHANACIMIENTO
9072  CC                 1 COLOMBIA        M           1945-09-14          
NA    NA                 NA                NA          <NA>                
46233 CC                 1 COLOMBIA        M           1944-12-13          
48486 CC                 1 COLOMBIA        F           1945-11-08          
49391 CC                 1 COLOMBIA        M           1946-07-27          
      PERIODO ESTU_CONSECUTIVO ESTU_ESTUDIANTE ESTU_ESTADOCIVIL
9072  20164   EK201640031759   ESTUDIANTE      CASADO          
NA    NA      NA               NA              NA              
46233 20164   EK201640037534   ESTUDIANTE      SOLTERO         
48486 20164   EK201640046175   ESTUDIANTE      SOLTERO         
49391 20164   EK201640001126   ESTUDIANTE      CASADO          
      ESTU_TIENE_ETNIA ESTU_ETNIA  ⋯ MOD_INGLES_DESEM MOD_INGLES_PNAL
9072     NA            NA          ⋯ A2               80             
NA       NA            NA          ⋯ NA               NA             
46233 FALSE            NA          ⋯ A1               52             
48486  TRUE            100 Ninguno ⋯ A1               50             
49391 FALSE            NA          ⋯ A2               63             
      MOD_INGLES_PGREF MOD_COMUNI_ESCRITA_PUNT MOD_COMUNI_ESCRITA_DESEM
9072  79                98                     2                       
NA    NA                NA                     NA                      
46233 72               119                     3                       
48486 50                85                     2                       
49391 51               104                     3                       
      MOD_COMUNI_ESCRITA_PNAL MOD_COMUNI_ESCRITA_PGREF ESTU_ESTADO INSE    
9072  39                      36                       PUBLICAR    53.16704
NA    NA                      NA                       NA                NA
46233 82                      85                       PUBLICAR    31.34326
48486 26                      26                       PUBLICAR    48.35082
49391 59                      62                       PUBLICAR    27.62424
      NSE 
9072  NSE2
NA    NA  
46233 NSE1
48486 NSE2
49391 NSE1

Para ejercicios practicos pondremos en NA las fechas de personas que tengan menos de 16 años.

In [70]:
df_published$ESTU_FECHANACIMIENTO[df_published$ESTU_FECHANACIMIENTO>as.Date("2000-12-31")] = NA

### Datos Númericos

In [71]:
numeric_colums = names(df_published)[sapply(df_published, is.numeric)]

In [72]:
setDT(df_published)[, summary(.SD), .SDcols = numeric_colums]

 MOD_RAZONA_CUANTITAT_PUNT MOD_RAZONA_CUANTITATIVO_PNAL
 Min.   :  0.00            Min.   :  1.00              
 1st Qu.: 81.00            1st Qu.: 26.00              
 Median : 96.00            Median : 51.00              
 Mean   : 96.48            Mean   : 50.53              
 3rd Qu.:111.00            3rd Qu.: 76.00              
 Max.   :200.00            Max.   :100.00              
                                                       
 MOD_RAZONA_CUANTITATIVO_PGREF MOD_LECTURA_CRITICA_PUNT
 Min.   :  1.00                Min.   :  0.0           
 1st Qu.: 26.00                1st Qu.: 81.0           
 Median : 51.00                Median : 97.0           
 Mean   : 50.53                Mean   : 95.8           
 3rd Qu.: 76.00                3rd Qu.:112.0           
 Max.   :100.00                Max.   :200.0           
                                                       
 MOD_LECTURA_CRITICA_PNAL MOD_LECTURA_CRITICA_PGREF MOD_COMPETEN_CIUDADA_PUNT
 Min.   :  1.00           

Tenemos que en el Modulo de Comunicación Escrita existen 770 valores nulos y que el puntaje minimo fue de 49 cuando en el resto de las pruebas los puntajes minimos suelen ser 0. Habría que revisar junto el icfes si hubo algún tipo de error en la recolección de esta variable.

No parece que hayan errores más allá de los datos nulos en el INSE y en la Competencia de Comunicación Escrita. Para estos NA no tenemos forma de calcular los valores nulos.

### Datos Cualitativos

In [81]:
is.not.numeric = function(x) {
    return(!is.numeric(x))
}

In [82]:
cualitative_colums = names(df_published)[sapply(df_published, is.not.numeric)]

In [84]:
setDT(df_published)[, summary(.SD), .SDcols = cualitative_colums]

 ESTU_TIPODOCUMENTO           ESTU_NACIONALIDAD ESTU_GENERO 
 CC:50358           1 COLOMBIA         :52665   F   :24667  
 CE:   36           112 JAPÓN          :    0   M   :27996  
 CR:  248           118 COREA DEL NORTE:    0   NA's:    2  
 PC:    1           119 COREA DEL SUR  :    0               
 PE:    0           12 ARGENTINA       :    0               
 TI: 2022           144 MÉXICO         :    0               
                    (Other)            :    0               
 ESTU_FECHANACIMIENTO  PERIODO            ESTU_CONSECUTIVO   ESTU_ESTUDIANTE 
 Min.   :1944-12-13   20164:52665   EK201640000012:    1   ESTUDIANTE:52665  
 1st Qu.:1988-04-20                 EK201640000014:    1                     
 Median :1993-08-03                 EK201640000015:    1                     
 Mean   :1991-04-13                 EK201640000017:    1                     
 3rd Qu.:1996-03-12                 EK201640000018:    1                     
 Max.   :2000-12-31                 EK201640

Hay varios valores NA en ESTU_GENERO, ESTU_ESTADOCIVIL, ESTU_TIENE_ETNIA y NSE  que no tenemos forma de llenarlos con los datos que se nos sumisnistrarón.

Podría ser posible calcular los MOD_COMUNI_ESCRITA_DESE faltantes, sin embargo esto requiere que el icfes nos de conocimiento de la forma en la que se calcula ya que no se encontró información pública de este calculo.

Por otro lado vemos que existen 49579 valores nulos en ESTU_ETNIA que se podría suponer que son personas sin etnia (aunque existe el campo ninguna), sin embargo solo 45098 afirmarón no tener etnia. Lo que nos lleva a pensar que hay inconsistencias en el llenado de las etnias.

### Analisis de Variables en Grupos

Existen variables que independientemente parecen tener sentido, sin embargo miremos el signifiado global de algunas de ellas para ver si existen inconsistencias como la que creemos que hay en las etnias.

#### Etnias

In [102]:
df_etnias = df_published[df_published$ESTU_TIENE_ETNIA == TRUE & df_published$ESTU_ETNIA == "100 Ninguno",]
select(df_etnias, c('ESTU_TIENE_ETNIA', 'ESTU_ETNIA'))

ESTU_TIENE_ETNIA ESTU_ETNIA 
1    TRUE             100 Ninguno
2    TRUE             100 Ninguno
3    TRUE             100 Ninguno
4    TRUE             100 Ninguno
5    TRUE             100 Ninguno
6    TRUE             100 Ninguno
7    TRUE             100 Ninguno
8    TRUE             100 Ninguno
9    TRUE             100 Ninguno
10   TRUE             100 Ninguno
11   TRUE             100 Ninguno
12   TRUE             100 Ninguno
13   TRUE             100 Ninguno
14   TRUE             100 Ninguno
15   TRUE             100 Ninguno
16   TRUE             100 Ninguno
17   TRUE             100 Ninguno
18   TRUE             100 Ninguno
19   TRUE             100 Ninguno
20   TRUE             100 Ninguno
21   TRUE             100 Ninguno
22   TRUE             100 Ninguno
23   TRUE             100 Ninguno
24   TRUE             100 Ninguno
25   TRUE             100 Ninguno
26   TRUE             100 Ninguno
27   TRUE             100 Ninguno
28   TRUE             100 Ninguno
29   TRUE             100 Ninguno
30   TRUE             100 Ninguno
⋮    ⋮                ⋮          
1561 TRUE             100 Ninguno
1562 TRUE             100 Ninguno
1563 TRUE             100 Ninguno
1564 TRUE             100 Ninguno
1565 TRUE             100 Ninguno
1566 TRUE             100 Ninguno
1567 TRUE             100 Ninguno
1568 TRUE             100 Ninguno
1569 TRUE             100 Ninguno
1570 TRUE             100 Ninguno
1571 TRUE             100 Ninguno
1572 TRUE             100 Ninguno
1573 TRUE             100 Ninguno
1574 TRUE             100 Ninguno
1575 TRUE             100 Ninguno
1576 TRUE             100 Ninguno
1577 TRUE             100 Ninguno
1578 TRUE             100 Ninguno
1579 TRUE             100 Ninguno
1580 TRUE             100 Ninguno
1581 TRUE             100 Ninguno
1582 TRUE             100 Ninguno
1583 TRUE             100 Ninguno
1584 TRUE             100 Ninguno
1585 TRUE             100 Ninguno
1586 TRUE             100 Ninguno
1587 TRUE             100 Ninguno
1588 TRUE             100 Ninguno
1589 TRUE             100 Ninguno
1590 TRUE             100 Ninguno

Existen 1590 persosas que afirmarón tener etnia y seleccionarón ninguna.

In [104]:
df_etnias = df_published[df_published$ESTU_TIENE_ETNIA == TRUE &  is.na(df_published$ESTU_ETNI),]
select(df_etnias, c('ESTU_TIENE_ETNIA', 'ESTU_ETNIA'))

ESTU_TIENE_ETNIA ESTU_ETNIA
1   TRUE             NA        
2   TRUE             NA        
3   TRUE             NA        
4   TRUE             NA        
5   TRUE             NA        
6   TRUE             NA        
7   TRUE             NA        
8   TRUE             NA        
9   TRUE             NA        
10  TRUE             NA        
11  TRUE             NA        
12  TRUE             NA        
13  TRUE             NA        
14  TRUE             NA        
15  TRUE             NA        
16  TRUE             NA        
17  TRUE             NA        
18  TRUE             NA        
19  TRUE             NA        
20  TRUE             NA        
21  TRUE             NA        
22  TRUE             NA        
23  TRUE             NA        
24  TRUE             NA        
25  TRUE             NA        
26  TRUE             NA        
27  TRUE             NA        
28  TRUE             NA        
29  TRUE             NA        
30  TRUE             NA        
⋮   ⋮                ⋮         
237 TRUE             NA        
238 TRUE             NA        
239 TRUE             NA        
240 TRUE             NA        
241 TRUE             NA        
242 TRUE             NA        
243 TRUE             NA        
244 TRUE             NA        
245 TRUE             NA        
246 TRUE             NA        
247 TRUE             NA        
248 TRUE             NA        
249 TRUE             NA        
250 TRUE             NA        
251 TRUE             NA        
252 TRUE             NA        
253 TRUE             NA        
254 TRUE             NA        
255 TRUE             NA        
256 TRUE             NA        
257 TRUE             NA        
258 TRUE             NA        
259 TRUE             NA        
260 TRUE             NA        
261 TRUE             NA        
262 TRUE             NA        
263 TRUE             NA        
264 TRUE             NA        
265 TRUE             NA        
266 TRUE             NA

Existen también 226 personas que afirmarón tener Etnia y no llenarón el campo de que etnia pertenecen.

In [106]:
df_etnias = df_published[df_published$ESTU_TIENE_ETNIA == FALSE & !is.na(df_published$ESTU_ETNI) & df_published$ESTU_ETNI != "100 Ninguno",]
select(df_etnias, c('ESTU_TIENE_ETNIA', 'ESTU_ETNIA'))

ESTU_TIENE_ETNIA ESTU_ETNIA                      
1  FALSE            19 Otro grupo étnico minoritario
2  FALSE            19 Otro grupo étnico minoritario
3  FALSE            1 Comunidad afrodescendiente    
4  FALSE            1 Comunidad afrodescendiente    
5  FALSE            9 Pasto                         
6  FALSE            19 Otro grupo étnico minoritario
7  FALSE            1 Comunidad afrodescendiente    
8  FALSE            19 Otro grupo étnico minoritario
9  FALSE            9 Pasto                         
10 FALSE            9 Pasto                         
11 FALSE            19 Otro grupo étnico minoritario
12 FALSE            19 Otro grupo étnico minoritario
13 FALSE            19 Otro grupo étnico minoritario
14 FALSE            19 Otro grupo étnico minoritario
15 FALSE            1 Comunidad afrodescendiente    
16 FALSE            19 Otro grupo étnico minoritario
17 FALSE            19 Otro grupo étnico minoritario
18 FALSE            19 Otro grupo étnico minoritario
19 FALSE            18 Zenú                         
20 FALSE            19 Otro grupo étnico minoritario
21 FALSE            19 Otro grupo étnico minoritario
22 FALSE            19 Otro grupo étnico minoritario
23 FALSE            19 Otro grupo étnico minoritario
24 FALSE            19 Otro grupo étnico minoritario
25 FALSE            1 Comunidad afrodescendiente    
26 FALSE            1 Comunidad afrodescendiente    
27 FALSE            7 Pijao                         
28 FALSE            2 Raizal                        
29 FALSE            19 Otro grupo étnico minoritario
30 FALSE            2 Raizal                        
31 FALSE            1 Comunidad afrodescendiente    
32 FALSE            19 Otro grupo étnico minoritario
33 FALSE            19 Otro grupo étnico minoritario
34 FALSE            1 Comunidad afrodescendiente    
35 FALSE            19 Otro grupo étnico minoritario

Hay 35 personas que dijeron no tener etnia pero seleccionarón alguna etnia.

Existe inconsistencias en el llenado de los datos, se recomendaría no preguntar si o no. Sino directamente al grupo en el que te pertenece con la opción de ninguno.

## Transformaciones

Ahora agregaremos los campos que se nos solicitó.

### Edad

Agregaremos la columna de edad a los datos.

In [113]:
age_cal = function (birthdate) {
    presentation_date = "2016-10-09"
    birthdate_lt = as.POSIXlt(birthdate)
    presentation_date_lt = as.POSIXlt(presentation_date)
    
    age = presentation_date_lt$year - birthdate_lt$year
    
    ifelse(presentation_date_lt$mon < birthdate_lt$mon |
         (presentation_date_lt$mon == birthdate_lt$mon & presentation_date_lt$mday < birthdate_lt$mday),
         age - 1, age)
}

In [118]:
df_published$ESTU_EDAD <- sapply(df_published$ESTU_FECHANACIMIENTO, age_cal)

In [120]:
df_published$ESTU_EDAD = as.integer(df_published$ESTU_EDAD)

In [121]:
df_published

ESTU_TIPODOCUMENTO ESTU_NACIONALIDAD ESTU_GENERO ESTU_FECHANACIMIENTO
1     CC                 1 COLOMBIA        M           1990-08-15          
2     CC                 1 COLOMBIA        M           1990-09-26          
3     CC                 1 COLOMBIA        F           1994-10-28          
4     CC                 1 COLOMBIA        F           1979-09-24          
5     CC                 1 COLOMBIA        M           1988-07-06          
6     CC                 1 COLOMBIA        M           1992-12-31          
7     CC                 1 COLOMBIA        F           1975-04-19          
8     CC                 1 COLOMBIA        M           1995-04-01          
9     CC                 1 COLOMBIA        M           1988-12-20          
10    CC                 1 COLOMBIA        F           1997-12-30          
11    CC                 1 COLOMBIA        M           1977-03-05          
12    CC                 1 COLOMBIA        F           1986-01-21          
13    CC                 1 COLOMBIA        F           1989-11-11          
14    CC                 1 COLOMBIA        F           1995-06-06          
15    CC                 1 COLOMBIA        M           1985-11-09          
16    CC                 1 COLOMBIA        F           1992-04-20          
17    CC                 1 COLOMBIA        F           1995-07-05          
18    CC                 1 COLOMBIA        M           1973-06-08          
19    CC                 1 COLOMBIA        M           1996-01-29          
20    CC                 1 COLOMBIA        M           1991-01-03          
21    CC                 1 COLOMBIA        F           1990-05-15          
22    CC                 1 COLOMBIA        M           1990-08-12          
23    CC                 1 COLOMBIA        M           1996-07-29          
24    TI                 1 COLOMBIA        F           1999-04-29          
25    CC                 1 COLOMBIA        M           1994-10-14          
26    CC                 1 COLOMBIA        F           1993-01-24          
27    CC                 1 COLOMBIA        M           1994-08-23          
28    CC                 1 COLOMBIA        F           1994-02-15          
29    CC                 1 COLOMBIA        M           1995-05-12          
30    CC                 1 COLOMBIA        M           1997-07-13          
⋮     ⋮                  ⋮                 ⋮           ⋮                   
52636 CC                 1 COLOMBIA        F           1993-12-10          
52637 CC                 1 COLOMBIA        M           1994-10-30          
52638 CC                 1 COLOMBIA        M           1996-05-04          
52639 TI                 1 COLOMBIA        M           1997-12-06          
52640 CC                 1 COLOMBIA        M           1984-10-18          
52641 CC                 1 COLOMBIA        M           1997-10-27          
52642 CC                 1 COLOMBIA        M           1995-07-17          
52643 CC                 1 COLOMBIA        M           1965-08-11          
52644 CC                 1 COLOMBIA        F           1993-01-23          
52645 CC                 1 COLOMBIA        F           1996-03-16          
52646 CC                 1 COLOMBIA        M           1983-10-11          
52647 CC                 1 COLOMBIA        F           1988-10-19          
52648 CC                 1 COLOMBIA        M           1993-11-13          
52649 CC                 1 COLOMBIA        M           1993-09-10          
52650 CC                 1 COLOMBIA        M           1983-05-12          
52651 CC                 1 COLOMBIA        M           1994-01-08          
52652 CC                 1 COLOMBIA        M           1994-02-12          
52653 CC                 1 COLOMBIA        F           1980-09-01          
52654 TI                 1 COLOMBIA        F           1999-01-26          
52655 CC                 1 COLOMBIA        F           1995-04-03          
52656 CC                 1 COLOMBIA        F          

### Nivel de Ingles

Agregaremos un campo que nos permita identificar si las personas llegarón a un nivel B1 o B2 de ingles.

In [130]:
alcanza_desempeño_ingles = lapply(df_published$MOD_INGLES_DESEM, function(x) { 
    if (x == "B1" | x == "B2") {
        return(TRUE) 
    } else {
        return(FALSE)
    }
})

In [132]:
df_published$ESTU_ALCANZA_COMPETENCIA_INGLES = as.logical(alcanza_desempeño_ingles

In [133]:
df_published

ESTU_TIPODOCUMENTO ESTU_NACIONALIDAD ESTU_GENERO ESTU_FECHANACIMIENTO
1     CC                 1 COLOMBIA        M           1990-08-15          
2     CC                 1 COLOMBIA        M           1990-09-26          
3     CC                 1 COLOMBIA        F           1994-10-28          
4     CC                 1 COLOMBIA        F           1979-09-24          
5     CC                 1 COLOMBIA        M           1988-07-06          
6     CC                 1 COLOMBIA        M           1992-12-31          
7     CC                 1 COLOMBIA        F           1975-04-19          
8     CC                 1 COLOMBIA        M           1995-04-01          
9     CC                 1 COLOMBIA        M           1988-12-20          
10    CC                 1 COLOMBIA        F           1997-12-30          
11    CC                 1 COLOMBIA        M           1977-03-05          
12    CC                 1 COLOMBIA        F           1986-01-21          
13    CC                 1 COLOMBIA        F           1989-11-11          
14    CC                 1 COLOMBIA        F           1995-06-06          
15    CC                 1 COLOMBIA        M           1985-11-09          
16    CC                 1 COLOMBIA        F           1992-04-20          
17    CC                 1 COLOMBIA        F           1995-07-05          
18    CC                 1 COLOMBIA        M           1973-06-08          
19    CC                 1 COLOMBIA        M           1996-01-29          
20    CC                 1 COLOMBIA        M           1991-01-03          
21    CC                 1 COLOMBIA        F           1990-05-15          
22    CC                 1 COLOMBIA        M           1990-08-12          
23    CC                 1 COLOMBIA        M           1996-07-29          
24    TI                 1 COLOMBIA        F           1999-04-29          
25    CC                 1 COLOMBIA        M           1994-10-14          
26    CC                 1 COLOMBIA        F           1993-01-24          
27    CC                 1 COLOMBIA        M           1994-08-23          
28    CC                 1 COLOMBIA        F           1994-02-15          
29    CC                 1 COLOMBIA        M           1995-05-12          
30    CC                 1 COLOMBIA        M           1997-07-13          
⋮     ⋮                  ⋮                 ⋮           ⋮                   
52636 CC                 1 COLOMBIA        F           1993-12-10          
52637 CC                 1 COLOMBIA        M           1994-10-30          
52638 CC                 1 COLOMBIA        M           1996-05-04          
52639 TI                 1 COLOMBIA        M           1997-12-06          
52640 CC                 1 COLOMBIA        M           1984-10-18          
52641 CC                 1 COLOMBIA        M           1997-10-27          
52642 CC                 1 COLOMBIA        M           1995-07-17          
52643 CC                 1 COLOMBIA        M           1965-08-11          
52644 CC                 1 COLOMBIA        F           1993-01-23          
52645 CC                 1 COLOMBIA        F           1996-03-16          
52646 CC                 1 COLOMBIA        M           1983-10-11          
52647 CC                 1 COLOMBIA        F           1988-10-19          
52648 CC                 1 COLOMBIA        M           1993-11-13          
52649 CC                 1 COLOMBIA        M           1993-09-10          
52650 CC                 1 COLOMBIA        M           1983-05-12          
52651 CC                 1 COLOMBIA        M           1994-01-08          
52652 CC                 1 COLOMBIA        M           1994-02-12          
52653 CC                 1 COLOMBIA        F           1980-09-01          
52654 TI                 1 COLOMBIA        F           1999-01-26          
52655 CC                 1 COLOMBIA        F           1995-04-03          
52656 CC                 1 COLOMBIA        F          